how-to-contribute.md

```diff
@@ -1,177 +1,198 @@
-*   Packaging easy-to-install packages (such as RPMs) for the myriad of possible configurations out there. (The project does not maintain anything but the basic .zip and .tar.gz packages, but anyone is welcome to build their own specific packages and announce them on the forrest-dev list)
-*   ... and there is just one other thing - don't forget to tell everyone who asks, how great Shiro is! The more people that know about and start to use Shiro, the larger the pool of potential contributors will be.
+* Assisting to improve documentation and the website.
+* Testing Shiro on various configurations and reporting back, especially Shiro's less-frequently-used features.
+* New samples for the 'shiro-sample' to concisely describe and demonstrate features. Such samples can also enable automated testing.
+* Debugging - producing reproducible test cases and/or finding causes of bugs. Most bugs are [recorded as issues](#HowtoContribute-procedure)).
+* Providing new use-cases and requirements.
+** If you think that Shiro does not quite meet your needs then tell us about it on the mailing list.
+* Specifying, analyzing or designing new features - and beyond.
+**	If you wish to get further involved with this, please join the [`shiro-dev` mailing list](mailing-lists.html "Mailing Lists"), install and try out Shiro and read some of the [mail archives](mailing-lists.html "Mailing Lists").
+***	Don't just say "it should have XYZ" without reading anything first - because chances are, somebody has already thought of that feature.
+* Packaging easy-to-install packages (such as Red Hat Package Managers (RPM)) for the myriad of possible configurations out there.
+** The project does not maintain anything but the basic .zip and .tar.gz packages, but anyone is welcome to build their own specific packages and announce those on the mailing list.
+* ... and one more thing - don't forget to tell everyone, how great Shiro is! The more people that know about and start to use Shiro, the larger the pool of potential contributors will be.
 
 <a name="HowtoContribute-procedure"></a>
 <a name="HowtoContribute-Procedureforreportingbugsandissuesandenhancementsuggestions"></a>
-##Procedure for reporting bugs and issues and enhancement suggestions
+## Procedure for Reporting Bugs and Issues and Enhancement Suggestions
 
-If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
+If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) first as it may be already reported.
```

Comment:
```suggestion
If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
```

no capital 'j', or dead link.